File: lua/problem_37.lua
local function powerSet(nums)
    local result = {{}}
    for _, v in ipairs(nums) do
        local n = #result
        for i = 1, n do
            local sub = {}
            for _, x in ipairs(result[i]) do sub[#sub+1] = x end
            sub[#sub+1] = v
            result[#result+1] = sub
        end
    end
    return result
end

local function problem37()
    print("Problem 37:")
    assert(#powerSet({1,2,3}) == 8, "assertion failed")
    assert(#powerSet({})      == 1, "assertion failed")
    assert(#powerSet({1})     == 2, "assertion failed")
    print("All Tests Passed")
end

return { problem37 = problem37 }

File: python/problem_37.py
from typing import List


def power_set(nums: List[int]) -> List[List[int]]:
    result = [[]]
    for v in nums:
        result += [subset + [v] for subset in result]
    return result


def main():
    print("Problem 37:")
    assert len(power_set([1, 2, 3])) == 8
    assert len(power_set([])) == 1
    assert len(power_set([1])) == 2
    print("All Tests Passed")
